weblate/api/serializers.py

```diff
@@ -502,6 +503,62 @@ class Meta:
         )
 
 
+class TermSerializer(serializers.ModelSerializer):
+    language = LanguageSerializer(required=False)
+
+    class Meta:
+        model = Term
+        fields = (
+            "language",
+            "id",
+            "source",
+            "target",
+        )
+
+    def to_internal_value(self, data):
+        result = super().to_internal_value(data)
+        if "language" in result:
+            result["language"] = Language.objects.get(
+                code=result["language"].get("code")
+            )
+        return result
+
+
+class GlossarySerializer(serializers.ModelSerializer):
+    project = ProjectSerializer(read_only=True)
+    source_language = LanguageSerializer(required=False)
+    projects_url = serializers.HyperlinkedIdentityField(
+        view_name="api:glossary-projects", lookup_field="id"
+    )
+    terms_url = serializers.HyperlinkedIdentityField(
+        view_name="api:glossary-terms", lookup_field="id"
+    )
+
+    class Meta:
+        model = Glossary
+        fields = (
+            "name",
+            "id",
+            "color",
+            "source_language",
+            "project",
+            "projects_url",
+            "terms_url",
+            "url",
+        )
+        extra_kwargs = {
+            "url": {"view_name": "api:glossary-detail", "lookup_field": "id"}
+        }
+
+    def to_internal_value(self, data):
+        result = super().to_internal_value(data)
+        if "source_language" in result:
+            result["source_language"] = Language.objects.get(
+                code=result["source_language"].get("code")
```

Comment:
The `get` invocation has no benefits here:

```suggestion
                code=result["source_language"]["code"]
```